src/component/header/components/BasketIcon.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { useCallback, useContext, useMemo } from 'react'
+import { useNavigate } from 'react-router';
+import IconButton from '@material-ui/core/IconButton';
+import ShoppingBasketIcon from '@material-ui/icons/ShoppingBasket';
+
+import { RouteNames } from '../../../interface/IRoute';
+import { BasketContext } from '../../../context/BasketContext';
+import { useStyles } from '../style';
+
+export const BasketIcon = () => {
+    const { booksLocalStore } = useContext(BasketContext)
+
+    const navigate = useNavigate()
+    const classes = useStyles()
+
+    const count = useMemo(() => 
+        booksLocalStore.map(book => +book.quantity).reduce((total: number, amount: number) => total + amount, 0)
```

Comment:
amount - rename to quantity

Reply:
booksLocalStore.reduce((total: number, book: IBook) => total + +book.quantity, 0)

Reply:
fixed